### DISCUSSION TITLE
Adding Word Bank / Adding Click Responses

### DISCUSSION THREAD

**Discussion by BobRoss048** [jspsych/jsPsych#1103] [Q&A]
Hello,

I'm attempting to create a simple reaction based task (e.g. see image, indicate what image is), but before participants see an image, they see a prime word (e.g. Truck, Car, etc).  I was wondering if there was a way to create an array of sorts & have it randomize the word they see.  I only want them to see a single word & then move on, but have the word be different every time - is this possible?  

Thank you!

**Comment by becky-gilbert** [comment]
For your first question, there are a few ways to do this. What I would do is (1) randomize the list of prime words, (2) loop through the the other trial information and add the corresponding randomized prime word to each trial, then (3) use this array of trial information to set up the trials using [timeline variables](https://www.jspsych.org/overview/timeline/#creating-an-experiment-the-timeline):
```javascript
    // create an empty timeline array
    var timeline = [];
    // create list of prime words to choose from
    var prime_word_list = ["Truck", "Car", "Van"];
    // shuffle the prime word order
    var prime_word_list_rand = jsPsych.randomization.shuffle(prime_word_list);
    // set up other trial information
    var trial_info = [
        {image: 'image1.png'},
        {image: 'image2.png'},
        {image: 'image3.png'}
    ];
    // for each object in the trial info array, add the corresponding prime word from the randomized list
    for (var i=0; i<trial_info.length; i++) {
        trial_info[i].prime_word = prime_word_list_rand[i];
    }
    var prime_trial = {
        type: 'html-keyboard-response',
        stimulus: jsPsych.timelineVariable('prime_word')
    };
    var image_trial = {
        type: 'image-keyboard-response',
        stimulus: jsPsych.timelineVariable('image')
    };
    var trial_procedure = {
        timeline: [prime_trial, image_trial],
        timeline_variables: trial_info
    };
    timeline.push(trial_procedure);
```
For the image click-as-response question - would you mind posting that as a separate discussion topic? We try to stick to one question per thread to make it easier for others to find answers by searching, and because it makes it easier to mark questions as answered. Thanks!

**Reply by BobRoss048** [reply]
Hello,

I edited my post to reflect a single question.  Thank you for bringing that to my attention.  I'll start a second post for the second question.  

As for the first part, I'll give this a shot!  Thank you so much.
